lib/OpenQA/Schema/ResultSet/JobSettings.pm

```diff
@@ -39,24 +39,16 @@ sub query_for_settings {
     my @conds;
     # Search into the following job_settings
     for my $setting (keys %$args) {
-        if ($args->{$setting}) {
-            # for dynamic self joins we need to be creative ;(
-            my $tname = 'me';
-            if (@conds) {
-                $tname = "siblings";
-                if (@joins) {
-                    $tname = "siblings_" . (int(@joins) + 1);
-                }
-                push(@joins, 'siblings');
-            }
-            my $setting_value = ($args->{$setting} =~ /^:\w+:/) ? {'like', "$&%"} : $args->{$setting};
-            push(
-                @conds,
-                {
-                    "$tname.key"   => $setting,
-                    "$tname.value" => $setting_value
-                });
-        }
+        next unless if $args->{$setting};
+        # for dynamic self joins we need to be creative ;(
+        my $tname = 'me';
+        my $setting_value = ($args->{$setting} =~ /^:\w+:/) ? {'like', "$&%"} : $args->{$setting};
+        push(
+            @conds,
+            {
+                "$tname.key"   => $setting,
+                "$tname.value" => $setting_value
+            });
     }
     return $self->search({-and => \@conds}, {join => \@joins});
```

Comment:
The new code doesn't populate `@joins` at all. Intuitively I would say that cannot be right. (And if it was right than `{join => \@joins}` can be removed completely.)

Reply:
Right. I removed all references to `@joins` because according to the coverage analysis we actually never call it in tests. And as explained in https://github.com/os-autoinst/openQA/pull/3999/commits/e0e2e6646bb5900b6c8d375a2ea7e675ce0b0ebe I did a manual test – which might be wrong – to check if it's used. Maybe some of you have an idea how we would trigger the code that adds joins?